integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSource.scala

```diff
@@ -217,22 +229,212 @@ class CarbonSource extends CreatableRelationProvider with RelationProvider
+  }
+
+  /**
+   * Checks if table exists by checking its schema file
+   * @param absoluteTableIdentifier
+   * @return Boolean
+   */
+  private def checkIfTableExists(absoluteTableIdentifier: AbsoluteTableIdentifier): Boolean = {
+    val carbonTablePath: CarbonTablePath = CarbonStorePath
+      .getCarbonTablePath(absoluteTableIdentifier)
+    val schemaFilePath: String = carbonTablePath.getSchemaFilePath
+    FileFactory.isFileExist(schemaFilePath, FileFactory.FileType.LOCAL) ||
+      FileFactory.isFileExist(schemaFilePath, FileFactory.FileType.HDFS) ||
+      FileFactory.isFileExist(schemaFilePath, FileFactory.FileType.VIEWFS)
+  }
+
+  /**
+   * If user wants to stream data from carbondata table source
+   * and if following conditions are true:
+   *    1. No schema provided by the user in readStream()
+   *    2. spark.sql.streaming.schemaInference is set to true
+   * carbondata can infer a table schema from a valid table path
+   * The schema inference is not mandatory, but good have.
    * When possible, this method should return the schema of the given `files`.  When the format
    * does not support inference, or no valid files are given should return None.  In these cases
    * Spark will require that user specify the schema manually.
    */
-  def inferSchema(
-    sparkSession: SparkSession,
-    options: Map[String, String],
-    files: Seq[FileStatus]): Option[StructType] = Some(new StructType().add("value", StringType))
+  override def inferSchema(
+                   sparkSession: SparkSession,
+                   options: Map[String, String],
+                   files: Seq[FileStatus]): Option[StructType] = {
+    Some(new StructType().add("value", StringType))
+    val path = options.get("path")
+    val tablePath: String = path match {
+      case Some(value) => value
+      case None => ""
+    }
+    // Check if table with given path exists
+    validateTable(tablePath)
+    val metastore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+    val carbonTableSchema: org.apache.carbondata.format.TableSchema =
+      getTableSchema(sparkSession: SparkSession, tablePath: String)
+    val columnnSchemaValues = carbonTableSchema.getTable_columns.asScala.sortBy(_.schemaOrdinal)
+
+    import scala.collection.mutable.ListBuffer
+    import scala.collection.JavaConverters._
+    val tableColumnNames = new ListBuffer[String]()
+    for (columnName <- columnnSchemaValues) {
+      tableColumnNames.append(columnName.column_name)
+    }
+    val tableColumnNamesList = tableColumnNames.toList
+
+    var columnDataTypes = new ListBuffer[String]()
+    for(columnDataType <- columnnSchemaValues) {
+      columnDataTypes.append(columnDataType.data_type.toString)
+    }
+    val tableColumnDataTypeList = columnDataTypes.toList
+
+    val inferredSchema: Option[StructType] = new Some(new StructType())
+    for (i <- tableColumnNamesList.indices) {
+      inferredSchema.get.add(tableColumnNamesList(i), tableColumnDataTypeList(i))
```

Comment:
It seems you can use `columnName.column_name` directly instead of `tableColumnNamesList(i)`